pkg/cloudprovider/azure/block_store.go

```diff
@@ -233,11 +250,19 @@ func (b *blockStore) DeleteSnapshot(snapshotID string) error {
 }
 
 func getFullDiskName(subscription string, resourceGroup string, diskName string) string {
-	return fmt.Sprintf("/subscriptions/%v/resourceGroups/%v/providers/Microsoft.Compute/disks/%v", subscription, resourceGroup, diskName)
+	return fmt.Sprintf(diskURIFormatString, subscription, resourceGroup, diskName)
 }
 
-func getFullSnapshotName(subscription string, resourceGroup string, snapshotName string) string {
-	return fmt.Sprintf("/subscriptions/%v/resourceGroups/%v/providers/Microsoft.Compute/snapshots/%v", subscription, resourceGroup, snapshotName)
+// parseFullSnapshotName takes a snapshot URI and returns the subscription ID,
+// resource group, and snapshot name, or an error if the URI does not match
+// the regexp.
+func parseFullSnapshotName(name string) (string, string, string, error) {
```

Comment:
How would you feel about returning this? I find anything returning more than 2 non-error values is usually confusing down the road...

```go
type SnapshotIdentifier struct {
  Subscription string
  ResourceGroup string
  Name string
}
```